content/Objects/Objects.q.md

```diff
@@ -0,0 +1,1016 @@
+
+ - syntax error, unexpected `'echo'` (T_ECHO) on line number 14
+
+ - syntax error, unexpected end of file, expecting `','` or `';'` on line number 14
+
+:::
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+    class Person 
+    {
+        public $name = "Diana";
+        
+        public function eat()
+        {
+            echo "This is an eat() method.";
+        }
+    }
+
+    $john = new Person();
+    $john->name = "Charles";
+    echo $johnname;
+?>
+```
+/// type=MS, answer=[1,3]
+
+In the statement `$john->name = "Charles";` on line 13, replace `$john` with `$charles`. Execute the program. What are the error messages?
+
+ - Undefined variable: `johnname` on line number 14
+
+ - syntax error, unexpected `'echo'` (T_ECHO) on line number 14
+
+ - Warning: Creating default object from empty value on line number 13
+
+ - syntax error, unexpected `'$charles'` (T_VARIABLE) on line number 13
+
+ - syntax error, unexpected end of file, expecting `','` or `';'` on line number 14
+
+:::
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+    class Person 
+    {
+        public $name = "Diana";
+        
+        public function eat()
+        {
+            echo "This is an eat() method.";
+        }
+    }
+
+    $john = new Person();
+    $charles->name = "Charles";
+    echo $johnname;
+?>
+```
+/// type=SS, answer=[2]
+
+On line 12, remove the assignment operator `=` after `$john`. Execute the program. What is the error message?
+
+ - syntax error, unexpected `'='` on line number 12
+
+ - syntax error, unexpected `'new'` (T_NEW) on line number 12
+
+ - Warning: Creating default object from empty value on line number 13
+
+ - syntax error, unexpected `'$charles'` (T_VARIABLE) on line number 13
+
+ - syntax error, unexpected end of file, expecting `','` or `';'` on line number 14
+
+:::
+
+
+/// type=CR, answer=[tests/Objects/CorrectMultipleErrorTest.php]
+
+Correct the code so that it outputs the string `Charles`.
+
+```php
+<?php
+    class Person 
+    {
+        public $name = "Diana";
+        
+        public function eat()
+        {
+            echo "This is an eat() method.";
+        }
+    }
+
+    $john  new Person();
+    $charles->name = "Charles";
+    echo $johnname;
+?>
+```
+
++++
+
+
++++
+
+### Part 4: Practice
+
+/// type=CR, answer=[tests/Objects/CreateNewObjectTest.php]
+
+Write a program that uses a `new` keyword to create an instance of a certain class. First, use a `class` keyword to declare a class named `Animal`. Within the curly braces `{}`, add a property definition of a class property `$type` with the value `Dog`. Next, add a method definition for `move()` method. Inside the `move()` method body, add an `echo` statement to display the string `Animals move from one place to another.`. After the class declaration, add a statement that creates the `$pet` object an instance of an `Animal` class. Then, add another statement that calls the `move()` method of the `$pet` object. Run the program to view the output.
```

Comment:
This would be easier to read and follow if displayed as an ordered list. A second Practice question of a similar nature would be good to reinforce the syntax for the student.

Reply:
I think this is a good idea. I will add another practice problem related to this. With regards to the ordered list display type for the instruction I will have to consult the devs if this can be rendered properly in the reader. If it is possible I can edit this instruction to let it appear in ordered list form. Thank you.